### PR TITLE
feat: #65 display messages on proposal page

### PR DESCRIPTION
**Pull request by ryanchristo**
Closes: #65

This pull request enables us to see messages attached to proposals using the existing review components. If a message is not supported, we fallback to displaying the message in json format.

I was unable to test all stake types due to the following issue: https://github.com/regen-network/groups-ui/issues/77

### CODE REVIEW

File: src/components/organisms/proposal-summary.tsx [hunk 1]
```diff
@@ -1,4 +1,13 @@
-import type { UIGroup, UIProposal, Vote, VoteOptionType } from 'types'
+import { Any } from '@haveanicedavid/regen-ts/types/codegen/google/protobuf/any'
```
Comment (haveanicedavid) [inline src/components/organisms/proposal-summary.tsx:1]: You should be able to import `Any` from `types` as well (it's re-exported in `golang.types.ts`

Comment (ryanchristo) [reply src/components/organisms/proposal-summary.tsx]: https://github.com/regen-network/groups-ui/pull/66/commits/f740345783fdcf2dbfac11c58f1c4ebd88030837

File: src/components/molecules/json-display.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { useColorModeValue } from 'hooks/chakra-hooks'
 
 import { Card, CardBody } from '@/atoms'
 
-export const JSONDisplay = ({ data }: { data: Record<string, unknown> }) => {
+export const JSONDisplay = ({ data }: { data: Record<any, unknown> }) => {
```
Comment (ryanchristo) [inline src/components/molecules/json-display.tsx:5]: This gives us a lint warning but seems important that we support any here.

File: src/components/organisms/proposal-summary.tsx [hunk 3]
```diff
@@ -89,3 +101,82 @@ export const ProposalSummary = ({
     </Card>
   )
 }
+
+// TODO: https://github.com/regen-network/regen-js/issues/71
+function renderMessage(msg: any, groupPolicyAddress: string) {
```
Comment (ryanchristo) [inline src/components/organisms/proposal-summary.tsx:106]: This also gives us a lint warning but we should have a specific type that works correctly and would be temporary.

File: src/components/molecules/json-display.tsx (hunk 2 above)
Comment (haveanicedavid) [reply src/components/molecules/json-display.tsx:5]: @ryanchristo occasionally an `unknown` type will work, however you can just
```tsx
// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
on the line before. Sometimes `any` is needed

File: src/components/organisms/proposal-summary.tsx (hunk 3 above)
Comment (haveanicedavid) [reply src/components/organisms/proposal-summary.tsx:106]: Same thing. feel free to disable. Alternatively, if these are the same type you can define that somewhere else and use it, like 
```tsx
// eslint-disable-next-line @typescript-eslint/no-explicit-any
type MsgAnyType = any
```
so you only need to disable the linter once.

Either works

**Review by haveanicedavid — APPROVED** [review]

**Pull request merged**